lib/AST/Module.cpp

```diff
@@ -331,8 +339,25 @@ SourceLookupCache::SourceLookupCache(const ModuleDecl &M) {
   }
 }
 
+void SourceLookupCache::addDelayedMacroExpansions() {
+  for (auto *MED : DelayedMacroExpansions) {
+    auto *rewritten = evaluateOrDefault(MED->getASTContext().evaluator,
+                                        ExpandMacroExpansionDeclRequest{MED},
+                                        nullptr);
+    if (rewritten)
+      for (auto node : rewritten->getElements())
+        if (auto *decl = node.dyn_cast<Decl *>())
+          addToUnqualifiedLookupCache(TinyPtrVector<Decl *>{decl}, false);
```

Comment:
The macro expansion could produce other macro expansion declarations, and we should make sure to account for those. It currently looks like `addToUnqualifiedLookupCache` could add to `DelayedMacroExpansions` while we're iterating over it. We'll either not see those added macro expansion declarations (because of the `clear()` below) or we'll crash if the vector gets reallocated.